packages/shared/src/hooks/useOnboardingModal.ts

```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { MainFeedPage } from '../components/utilities';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import { Alerts } from '../graphql/alerts';
+import { OnboardingMode } from '../graphql/feed';
+import { OnboardingVersion } from '../lib/featureValues';
+import { LoggedUser } from '../lib/user';
+import usePersistentContext from './usePersistentContext';
+
+interface UseOnboardingModal {
+  myFeedMode: OnboardingMode;
+  isOnboardingOpen: boolean;
+  isLegacyOnboardingOpen: boolean;
+  onCloseOnboardingModal: () => void;
+  onInitializeOnboarding: () => void;
+}
+
+interface UseOnboardingModalProps {
+  user?: LoggedUser;
+  alerts: Alerts;
+  isFirstVisit: boolean;
+  onboardingVersion: OnboardingVersion;
+  onFeedPageChanged: (page: MainFeedPage) => unknown;
+}
+
+type ModalCommand = Record<OnboardingVersion, (value: boolean) => unknown>;
+
+const FIRST_TIME_SESSION = 'firstTimeSession';
+const LOGGED_USER_ONBOARDING = 'loggedUserOnboarding';
+
+export const useOnboardingModal = ({
+  user,
+  alerts,
+  isFirstVisit,
+  onboardingVersion,
+  onFeedPageChanged,
+}: UseOnboardingModalProps): UseOnboardingModal => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [onboardingMode, setOnboardingMode] = useState(OnboardingMode.Manual);
+  const [isFirstSession, setIsFirstSession, isSessionLoaded] =
```

Comment:
I'm not sure we need the first session tracking anymore. All we care about is whether the user has tried onboarding or has configured feed already

Reply:
Cool. In that case, I will remove this part and use a single variable only.